packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts

```diff
@@ -196,6 +196,14 @@ export interface NetworkLoadBalancedTaskImageProps {
    * @default - No labels.
    */
   readonly dockerLabels?: { [key: string]: string };
+  /**
+   * The amount (in GiB) of ephemeral storage to be allocated to the task. The maximum supported value is 200GiB.
+   *
+   * NOTE: This parameter is only supported for tasks hosted on AWS Fargate using platform version 1.4.0 or later.
+   *
+   * @default - Undefined, in which case, the task will receive 20GiB ephemeral storage.
+   */
+  readonly ephemeralStorageGiB?: number;
```

Comment:
Would it make sense to put this property on the `Fargate` interfaces rather than
the base interfaces? Then you wouldn't need to checks under the `ecs` services.

Reply:
Actually we can do that. But Where we closely observed the `taskImageOptions` were using in both fargate and ecs service patterns except in QueueProcessing. So I thought that its better to use the above property in the base interface rather than Fargate interfaces and pass the validation for ecs service patterns that when a customer is trying to use ephemeral storage by using ecs service pattern then it throws error saying "ephemeralStorageGiB is only supported for Fargate services."

When you see for  QueueProcessing where I put above property in the Fargate interfaces rather than
the base interfaces only. Because this was not using TaskImageOptions.

Reply:
Yeah I think we should now have separate taskImageOptions for ecs vs fargate since the properties that can be input are now different. From a user experience perspective it's better to not be able to input invalid properties than to get an error saying that you can't use a particular property.

Reply:
OK so do I need to remove that validation in ecs service patterns and add this property directly into fargate service patterns?